src/rpc/blockchain.cpp

```diff
@@ -793,7 +793,8 @@ static RPCHelpMan getblockfrompeer()
         "\nAttempt to fetch block from a given peer.\n"
         "\nWe must have the header for this block, e.g. using submitheader.\n"
         "Subsequent calls for the same block and a new peer will cause the response from the previous peer to be ignored.\n"
-        "\nReturns an empty JSON object if the request was successfully scheduled.",
+        "\nReturns an empty JSON object if the request was successfully scheduled."
+        "\nNote: On a pruned node this block might be pruned again as soon as the pruneheight surpasses the blockheight at the time of fetching the block.\n",
```

Comment:
What block does "the block" refer to? The most recent block that's being fetched as part of the regular activity?

Maybe this can be a bit shorter: "Previously pruned blocks are only preserved until the next pruning event." (afaik this is true as long as the user doesn't change `-prune`)

Reply:
"block" both times refers to the block being fetched with `getblockfrompeer`. I see how it was not super clear so I revised the wording a bit.

I think your shorter version is not necessarily correct. The block might be pruned with the next pruning event or not, it depends on the prune height requested if it is in a file that is otherwise ready to be pruned. So the block with the highest height in the same file is the deciding factor as far as I understand. Since the fetched blocked is usually appended in the file with the current tip, the next pruning event will often not allow that file to be pruned. The expanded test that you requested actually shows this well already and so I structured it in a way that this behavior is made explicit.

Reply:
Ah I see. Then I think there should be a warning that this can temporarily cause pruning to be deferred?

Reply:
Yes, if the node has not synced up to the fetched block yet, that can happen. That is the issue I am addressing with #23927 . I could amend the comment here now and then change it again if that PR is merged. But I think I would rather keep it separate since I also don't know what might be merged first. If the fix in #23927 is not getting in, I would change that PR to at least document the issue better for the user instead.